src/DataRetriever/LogDataRetriever.php

```diff
@@ -70,9 +70,13 @@ public function getLog(
     private function grepData(array $data, string $grepKeyword, bool $invertGrep): array
     {
         if (false === $invertGrep) {
-            return preg_grep("/$grepKeyword/", $data);
+            $lines = preg_grep("/$grepKeyword/", $data);
+
+            return \is_array($lines) ? $lines : [];
         }
 
-        return preg_grep("/$grepKeyword/", $data, \PREG_GREP_INVERT);
+        $lines = preg_grep("/$grepKeyword/", $data, \PREG_GREP_INVERT);
+
+        return \is_array($lines) ? $lines : [];
```

Comment:
What about something like this : 

```
    private function grepData(...) 
    {
        $lines = preg_grep("/$grepKeyword/", $data, $invertGrep ? \PREG_GREP_INVERT : null);

        return \is_array($lines) ? $lines : [];
    }
```

Reply:
Yep, i just change `null` to 0 because of `Parameter #3 $flags of function preg_grep expects int, int|null given.`, which 0 is the default value without there being a constant for this value.
https://www.php.net/manual/en/function.preg-grep.php